catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectSaveTask.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.io.asynctask;
+
+import android.os.AsyncTask;
+
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.io.XstreamSerializer;
+
+import java.lang.ref.WeakReference;
+
+public class ProjectSaveTask extends AsyncTask<Project, Void, Boolean> {
+
+	private WeakReference<ProjectSaveListener> weakListenerReference;
+
+	public ProjectSaveTask(ProjectSaveListener listener) {
+		weakListenerReference = new WeakReference<>(listener);
+	}
+
+	public static boolean task(Project project) {
+		return XstreamSerializer.getInstance().saveProject(project);
+	}
+
+	@Override
+	protected Boolean doInBackground(Project... projects) {
+		return task(projects[0]);
```

Comment:
because you can only save one project at a time, while you can import multiple projects within the same scope.

why don't we allow to save multiple projects at once you may ask, that's a good question, I'd reply. however, the `XstreamSerializer` is quite peculiar when it comes to loading and saving projects (i.e. we haven't yet refactored it) and the what the methods used for importing do on the other hand is clearer (i.e.we recently wrote them) so we are more confident that importing multiple projects is less likely to blow up than saving. besides I did not really see a use case where I'd want to save multiple projects at the same time, as our app only supports one loaded project (that can be modified) at a time anyway.